client/src/unifyfs-stdio.c

```diff
@@ -704,7 +704,7 @@ static int unifyfs_stream_write(
             errno = EBADF;
             return UNIFYFS_ERROR_BADF;
         }
-        current = unifyfs_fid_size(fid);
+        current = unifyfs_fid_local_size(fid);
```

Comment:
Let's change this to your logical_size() call to future proof a bit.  I think there's a good chance that we'll end up having to deal with the global file size before lamination at some point and the logical_size() call is a good abstraction for that.

Reply:
ok